src/Polly.Core/Simmy/Outcomes/OutcomeStrategyOptions.TResult.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace Polly.Simmy.Outcomes;
+
+#pragma warning disable CS8618 // Required members are not initialized in constructor since this is a DTO, default value is null
+
+/// <summary>
+/// Represents the options for the Behavior chaos strategy.
+/// </summary>
+/// <typeparam name="TResult">The type of the outcome that was injected.</typeparam>
+public class OutcomeStrategyOptions<TResult> : MonkeyStrategyOptions
+{
+    /// <summary>
+    /// Gets or sets the delegate that's raised when the outcome is injected.
+    /// </summary>
+    /// <remarks>
+    /// Defaults to <see langword="null"/>.
+    /// </remarks>
+    public Func<OnOutcomeInjectedArguments<TResult>, ValueTask>? OnOutcomeInjected { get; set; }
+
+    /// <summary>
+    /// Gets or sets the outcome generator to be injected for a given execution.
+    /// </summary>
+    /// <remarks>
+    /// Defaults to <see langword="null"/>. Either <see cref="Outcome"/> or this property is required.
+    /// When this property is <see langword="null"/> the <see cref="Outcome"/> is used.
+    /// </remarks>
+    public Func<ResilienceContext, ValueTask<Outcome<TResult>>> OutcomeGenerator { get; set; }
+
+    /// <summary>
+    /// Gets or sets the outcome to be injected for a given execution.
+    /// </summary>
+    /// <remarks>
+    /// Defaults to <see langword="null"/>. Either <see cref="OutcomeGenerator"/> or this property is required.
+    /// When this property is <see langword="null"/> the <see cref="OutcomeGenerator"/> is used.
+    /// </remarks>
+    public Outcome<TResult> Outcome { get; set; }
```

Comment:
```suggestion
    [Required]
    public Outcome<TResult>? Outcome { get; set; }
```

Reply:
It's not really required tho, since the consumer can either pass the `Outcome` or the `OutcomeGenerator`. I made it optional, good catch.